core/services/versionchooser/test_versionchooser.py

```diff
@@ -0,0 +1,193 @@
+        "image": "bluerobotics/companion-core",
+        "enabled": true,
+        "webui": false,
+        "network": "host",
+        "binds": {
+            "/dev/": {
+                "bind": "/dev/",
+                "mode": "rw"
+            },
+            "/var/run/wpa_supplicant/wlan0": {
+                "bind": "/var/run/wpa_supplicant/wlan0",
+                "mode": "rw"
+            },
+            "/tmp/wpa_playground": {
+                "bind": "/tmp/wpa_playground",
+                "mode": "rw"
+            }
+        },
+        "privileged": true
+    }
+}"""
+
+SAMPLE_IMAGE = json.loads(
+    """{
+   "attrs":{
+      "date":"2021-04-09T17:51:18.065721638Z"
+   },
+   "id":"856fdf5e66c9b3697c25015556e7895c9066febb1a8ac8657a4eb41f2fc95a57"
+}"""
+)
+
+
+@pytest.mark.asyncio
+async def test_get_version() -> None:
+    """Tests if VersionChooser.get_version is reading SAMPLE_JSON properly
+
+    Interacts with:
+        - docker client (get images)
+        - Settigns file
+    """
+    client_mock = mock.MagicMock()
+    chooser = VersionChooser(client_mock)
+
+    attrs = {
+        "images.get.return_value.id": "856fdf5e66c9b3697c25015556e7895c9066febb1a8ac8657a4eb41f2fc95a57",
+        "images.get.return_value.attrs.__getitem__.return_value": {"date": "2021-04-09T17:51:18.065721638Z"},
+    }
+    client_mock.configure_mock(**attrs)
+
+    # Mock so it doesn't try to read a real file from the filesystem
+    with mock.patch("builtins.open", mock.mock_open(read_data=SAMPLE_JSON)):
+
+        response = await chooser.get_version()
+        result = json.loads(response.text)
+        assert result["image"] == "bluerobotics/companion-core"
+        assert result["tag"] == "master"
+        assert len(client_mock.mock_calls) > 0
+
+
+version = {"tag": "master", "image": "bluerobotics/companion-core", "pull": False}
+
+
+@pytest.mark.asyncio
+@mock.patch("aiohttp.web.StreamResponse.write", new_callable=AsyncMock)
+async def test_set_version(write_mock: AsyncMock) -> None:
+    client = mock.MagicMock()
+    chooser = VersionChooser(client)
+    with mock.patch("builtins.open", mock.mock_open(read_data=SAMPLE_JSON)):
+
+        result = await chooser.set_version("bluerobotics/companion-core", "master")
+        assert await write_mock.called_once_with(
+            '{  "core": {\n    "tag": "master",\n    "image": "bluerobotics/companion-core",\n    "enabled": true,\n  '
+            '  "webui": false,\n    "network": "host",\n    "binds": {\n      "/dev/": {\n        "bind": "/dev/",\n  '
+            '      "mode": "rw"\n      },\n      "/var/run/wpa_supplicant/wlan0": {\n        "bind": "/var/run/wpa_sup'
+            'plicant/wlan0",\n        "mode": "rw"\n      },\n      "/tmp/wpa_playground": {\n        "bind": "/tmp/wp'
+            'a_playground",\n        "mode": "rw"\n      }\n    },\n    "privileged": true\n  }\n}'
+        )
+        assert result.status == 200
+        assert len(write_mock.mock_calls) > 0
```

Comment:
this was not required, since with already have called_once_with a few lines above, which makes sure the mock was used and receved the expected arguments. I took this line out.